tests/geomodel/core/test-geomodel-build.cpp

```diff
@@ -0,0 +1,641 @@
+// 7 Surfaces
+const index_t nb_surfaces = 7;
+// stergd Regions
+const index_t nb_regions = 2;
+
+bool test_build_geomodel( const GeoModel3D& geomodel )
+{
+    const index_t nb_surface_entities =
+        geomodel.nb_mesh_entities( surface_type_name_static() );
+    const index_t nb_region_entities =
+        geomodel.nb_mesh_entities( region_type_name_static() );
+    const index_t nb_corner_entities =
+        geomodel.nb_mesh_entities( corner_type_name_static() );
+    const index_t nb_line_entities =
+        geomodel.nb_mesh_entities( line_type_name_static() );
+
+    const index_t nb_contact_entities =
+        geomodel.nb_geological_entities( Contact3D::type_name_static() );
+    const index_t nb_interface_entities =
+        geomodel.nb_geological_entities( Interface3D::type_name_static() );
+    const index_t nb_layer_entities =
+        geomodel.nb_geological_entities( Layer3D::type_name_static() );
+
+    if( nb_contacts != nb_contact_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of contacts are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_interfaces != nb_interface_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of interfaces are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_layers != nb_layer_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of layers are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_surfaces != nb_surface_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of surfaces are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_regions != nb_region_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of regions are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_lines != nb_line_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of lines are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_corners != nb_corner_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of corners are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+    return true;
+}
+
+gmge_id create_geology_entity(
+    const GeologicalEntityType& entity_type, GeoModelBuilder3D& builder )
+{
+    return builder.geology.create_geological_entity( entity_type );
+}
+
+gmme_id create_topology_entity(
+    const MeshEntityType& entity_type, GeoModelBuilder3D& builder )
+{
+    return builder.topology.create_mesh_entity( entity_type );
+}
```

Comment:
What is the purpose of these two functions? Why not directly call builder.geology.create_geological_entity( entity_type ); and builder.topology.create_mesh_entity( entity_type );

Reply:
initially builder.geology.create_geological_entity( entity_type ); and builder.topology.create_mesh_entity( entity_type ); were directly called but as an exercice that used my c++ training I had to create those two fonctions.